docs/custom-types.md

```diff
@@ -106,20 +105,29 @@ config:
   hour: 1000
   limit_by: ip
   second: 100
+plugin: rate-limiting
 ```
 
 ### KongIngress
 
-This option allows us to configure setting from kong related to the [Upstream][5], [Service][6] and [routes][7] that are defined in the Kubernetes Ingress specification.
+Ingress resource spec in Kubernetes can defining routing policies based on HTTP Host header and paths.
+While this is sufficient in most cases, sometimes, users want more control over routing their at the Ingress level.
+`KongIngress` works in conjunction with existing Ingress resource and extends it. It is not meant as a replacement to the
+`Ingress` resource in Kubernetes.
+Using `KongIngress`, all properties of [Upstream][5], [Service][6] and [Route][7] entitise in Kong related to an Ingress resource
+can be modified.
 
-*There are two ways of activating this feature:*
-- We can create a `KongIngress` object in the same namespace of the Ingress rule using the same name. With this convention we can avoid an additional annotation in the Ingress. This is useful but requires one `KongIngress` definition per Ingress, which could become hard to maintain when have multiple rules. 
+Once a `KongIngress` resource is created, it can be associated with `Ingress` resource in two ways:
+- Create a `KongIngress` object in the same namespace as that of the Ingress rule using the same name.
+  This avoids a need of additional annotation in Ingress oresource.
+  On the other hand, this approach requires a `KongIngress` resource per Ingress, which becomes hard to maintain with multiple Ingresses.
 
-- (Alternatively) To simplify maintenance, by reusing the same KongIngress on multiple Ingress resources, we can create just one or more `KongIngress` object and map which one should be used using the annotation `configuration.konghq.com: <name>`.
+- Create an `KongIngress` resource and then using the annotation `configuration.konghq.com: <KongIngress-resource-name>`,
```

Comment:
Can we use an indent on this step since it's a substep?

Reply:
Umm, this is the same line? Not sure if I understand you correctly.

Reply:
Sorry bad eyes : )